### ISSUE TITLE
Mac Runtime compiles, but produces invalid code

### ISSUE THREAD

**Issue by ctel** [OliverSch/compiler#103]
Need some explanation, how to fix this; tried to fix it already, but on fist sight, it seems that mac uses some other way of calling...

**Comment by ctel** [comment]
.section	__TEXT,__text,regular,pure_instructions
	.globl	_print_int
	.align	4, 0x90
_print_int:                             ## @print_int
	.cfi_startproc
## BB#0:
	pushq	%rbp
Ltmp2:
	.cfi_def_cfa_offset 16
Ltmp3:
	.cfi_offset %rbp, -16
	movq	%rsp, %rbp
Ltmp4:
	.cfi_def_cfa_register %rbp
	subq	$16, %rsp
	leaq	L_.str(%rip), %rax
	movl	%edi, -4(%rbp)
	movl	-4(%rbp), %esi
	movq	%rax, %rdi
	movb	$0, %al
	callq	_printf
	movl	%eax, -8(%rbp)          ## 4-byte Spill
	addq	$16, %rsp
	popq	%rbp
	retq
	.cfi_endproc

	.globl	_calloc_wrapper
	.align	4, 0x90
_calloc_wrapper:                        ## @calloc_wrapper
	.cfi_startproc
## BB#0:
	pushq	%rbp
Ltmp7:
	.cfi_def_cfa_offset 16
Ltmp8:
	.cfi_offset %rbp, -16
	movq	%rsp, %rbp
Ltmp9:
	.cfi_def_cfa_register %rbp
	subq	$16, %rsp
	movq	%rdi, -8(%rbp)
	movq	%rsi, -16(%rbp)
	movq	-8(%rbp), %rdi
	movq	-16(%rbp), %rsi
	callq	_calloc
	addq	$16, %rsp
	popq	%rbp
	retq
	.cfi_endproc

	.section	__TEXT,__cstring,cstring_literals
L_.str:                                 ## @.str
	.asciz	"%i\n"


.subsections_via_symbols

**Comment by felixwilhelm** [comment]
Jap. OS X Runtime ist seit vorletzter Woche broken. Entweder ne Linux VM aufsetzen oder den linux runtime code nehmen.

**Comment by ctel** [comment]
Ja, ich dachte mir, das kann ich fixen. Linux runtime baut nicht und die VM krieg ich nicht zum Laufen. Ich wundere mich auch schon ne Weile darüber, warum der Code bei mir nicht zu den richtigen Ergebnissen führt, und bin bisher nicht drauf gekommen warum. Also dachte ich mir, dass das zu fixen vmtl das Einfachste ist... Oder?

**Comment by ctel** [comment]
Auf was muss ich denn Aufpassen wenn ich mich da dran wage? Welche Register sind denn zu berücksichtigen?

**Comment by felixwilhelm** [comment]
An sich kannst du einfach die linux runtime komplett übernehmen, musst halt potentiell die Syntax anpassen die nicht supported wird.

**Comment by ctel** [comment]
Okay, I'm on it!

**Comment by ctel** [comment]
Boah. Wie debugge ich da am Besten Segfaults? Eins zu eins mit den Änderungen kann mans nicht einfach übernehmen... Hab jetzt ein bisschen gebastelt und jetzt sieht es an sich gut aus, sodass der Compilevorgang wenigstens abschließt, aber die Produkte Segfaulten ständig und es lässt sich irgendwie nicht gescheit debuggen. Ne Idee?

**Comment by sweetpi** [comment]
gdb?

**Comment by OliverSch** [comment]
gdb?

**Comment by ctel** [comment]
Ich werd daraus halt nicht schlau und habe da kaum Erfahrung... Ich lad mein Zeug mal hoch...
